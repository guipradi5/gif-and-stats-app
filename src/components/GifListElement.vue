<template>
  <li @click="emit('select', element)">{{ element.content_description }}</li>
</template>

<script setup>
import { ref, defineProps, defineEmits } from "vue";

const props = defineProps(["element"])
const element = ref(props.element)

const emit = defineEmits(["select"])
</script>

<script>
export default {
  name: "GifListElement"
}
</script>

<style scoped lang="scss">
li {
  padding: 10px 20px;
  list-style: none;
  cursor: pointer;
}
</style>
