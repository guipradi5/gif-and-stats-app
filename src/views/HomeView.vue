<template>
  <main>  
    <aside class="side-bar">
      <h2>Top 8 gifs</h2>
      <ul class="side-bar__gif-list">
        <GifListElement v-for="element in gifs" :key="element.id" :element="element"></GifListElement>
      </ul>
    </aside>
  </main>
</template>

<script setup>
import { onMounted, ref } from 'vue'
import { getGifs } from '../api/api'
import GifListElement from '@/components/GifListElement.vue';

let gifs = ref([{id:1}])

onMounted(async () => {
  gifs.value = await getGifs()
})

</script>

<script>
export default {
  name: 'MainComponent'
}
</script>

<style scoped lang="scss">
.side-bar{
  max-width: 30%;
  border-right: white 1px solid;
  h2{
  padding: 15px;
  margin: 0;
  }
  &__gif-list{
    margin: 0;
    padding: 0;
  }
}
</style>
