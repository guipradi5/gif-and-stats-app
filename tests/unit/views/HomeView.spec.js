import { shallowMount } from "@vue/test-utils";
import HomeView from "@/views/HomeView.vue";
import GifListElement from "@/components/GifListElement";

jest.mock("@/api/api", () => ({
  ...jest.requireActual("@/api/api"),
  getGifs: () => [
    { id: 5, description: "gif" },
    { id: 6, description: "gif2" },
    { id: 7, description: "gif3" },
  ],
  getGifClickCount: () => 15,
}));

const component = () => {
  return shallowMount(HomeView, {});
};

describe("HomeView.vue", () => {
  it("HomeView should mount", () => {
    const wrapper = component();
    expect(wrapper.vm).toBeTruthy();
  });

  it("HomeView should render", async () => {
    const wrapper = await component();
    await wrapper.vm.$nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("HomeView should render", async () => {
    const wrapper = await component();
    await wrapper.vm.$nextTick();
    await wrapper
      .findAllComponents(GifListElement)
      .at(1)
      .trigger("select", { id: 6, description: "gif2" });
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.selectedGif.clickCount).toBe(15);
  });
});
