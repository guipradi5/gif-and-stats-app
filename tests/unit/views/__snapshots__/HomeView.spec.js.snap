// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView.vue HomeView should render 1`] = `
<main>
  <aside class="side-bar">
    <h2>Top 8 gifs</h2>
    <ul class="side-bar__gif-list">
      <gif-list-element-stub element="[object Object]"></gif-list-element-stub>
      <gif-list-element-stub element="[object Object]"></gif-list-element-stub>
      <gif-list-element-stub element="[object Object]"></gif-list-element-stub>
    </ul>
  </aside>
  <section class="main-body">
    <div class="gif-wrapper">
      <!--v-if-->
    </div>
  </section>
</main>
`;
